### PR TITLE
Added hot reload and access log to console

### PR DESCRIPTION
**Pull request by fractalzombie**
Hot reload added, but you need to install [https://github.com/emcrisostomo/fswatch](url). This package is cross-platform.

Added access log to console.

Some refactoring.

Happy New Year)))

### CODE REVIEW

**Comment by fractalzombie** [comment]
Huh... I fix it... laravel helpers is hell.

**Comment by albertcht** [comment]
Happy new year~

File: src/HotReload/FSProcess.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+     * Make swoole process.
+     *
+     * @param callable|null $callback
+     *
+     * @return \Swoole\Process
+     */
+    public function make(?callable $callback = null)
+    {
+        $mcb = function ($type, $buffer) use ($callback) {
+            if (! $this->locked && AppProcess::OUT === $type && $event = FSEventParser::toEvent($buffer)) {
```
Comment (albertcht) [inline src/HotReload/FSProcess.php]: I'm not sure why the event I got here is always `null`. So the hot reload didn't get triggered after receiving a fswatch event. Do you have any ideas?

**Comment by fractalzombie** [comment]
I will see, what happens. What is you use?

File: src/HotReload/FSProcess.php (hunk 1 above)
Comment (fractalzombie) [reply src/HotReload/FSProcess.php]: I will see, what happens. What is you use?

File: src/HotReload/FSEventParser.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+namespace SwooleTW\Http\HotReload;
+
+use Carbon\Carbon;
+
+/**
+ * Class FSEventParser
+ */
+class FSEventParser
+{
+    protected const REGEX = '/^([\S+]{3}\s+[\S+]{3}\s+[\d+]{2}\s+[\d+]{2}:[\d+]{2}:{0,2}:[\d+]{2}:{0,2}\s+[\d+]{0,4})\s+(\/[\S+]*)\s+([\S+*\s+]*)/mi';
```
Comment (albertcht) [inline src/HotReload/FSEventParser.php]: I found the reason that `FSEventParser::toEvent($event)` is always `null` in my environment. I'm using OSX and `fswatch 1.14.0`. Here's what I received for the fswatch event:

```
Mon Jan  7 23:30:28 2019 /Users/Albert/Projects/laravel-swoole-demo/vendor/swooletw/laravel-swoole/src/HotReload/FSProcess.php PlatformSpecific Updated IsFile
```

I haven't tested it on other platforms. But it works for my after changing the regex to:

```
^([\S+]{3}\s+[\S+]{3}\s+[\d+]{1,2}\s+...
```

File: src/HotReload/FSProcess.php (hunk 1 above)
Comment (albertcht) [reply src/HotReload/FSProcess.php]: I'm using OSX and `fswatch 1.14.0`.

Comment (fractalzombie) [reply src/HotReload/FSProcess.php]: > I'm using OSX and `fswatch 1.14.0`.

I'm too...

I fix it.

File: src/HotReload/FSEventParser.php (hunk 2 above)
Comment (fractalzombie) [reply src/HotReload/FSEventParser.php]: Fixed

Comment (albertcht) [reply src/HotReload/FSEventParser.php]: Hi @fractalzombie ,

Thank you so much ^^

**Pull request merged**